object/check.go

```diff
@@ -382,6 +382,11 @@ func CheckUserPassword(organization string, username string, password string, la
 			return nil, err
 		}
 	}
+	if CheckPasswordExpired(user) {
+		user.NeedUpdatePassword = true
```

Comment:
Don't mess with NeedUpdatePassword if you don't fully understand and know how to use it